dfdx-macros/src/reset_params.rs

```diff
@@ -0,0 +1,144 @@
+// Helper functions for the ResetParams derive macro
+use darling::{Error, FromDeriveInput, FromMeta, Result};
+use proc_macro2::TokenStream;
+use quote::{quote, quote_spanned};
+use syn::spanned::Spanned;
+
+#[derive(Debug)]
+enum Initializers {
```

Comment:
I think for now we can use this ResetParams for only handling modules that just call reset on all sub modules. Things that have actual logic in reset params can just implement by hand